v2/reader_test.go

```diff
@@ -178,6 +178,27 @@ func TestReader_WithCarV2Consistency(t *testing.T) {
 	}
 }
 
+func TestOpenReader_DoesNotPanicAfterClosure(t *testing.T) {
+	subject, err := carv2.OpenReader("testdata/sample-wrapped-v2.car")
+	require.NoError(t, err)
+	dReaderBeforeClosure := subject.DataReader()
+	iReaderBeforeClosure := subject.IndexReader()
+	require.NoError(t, subject.Close())
+	dReaderAfterClosure := subject.DataReader()
```

Comment:
@masih Can you try this without using the new readers here ? As in use the `dReaderBeforeClosure ` and `iReaderBeforeClosure ` created before we closed the file.

Reply:
Sure

Reply:
Separated the tests [here](https://github.com/ipld/go-car/pull/212/commits/912af7dd665ee51430194a6533308a5b9257aa0c) for before and after.